lib/aerospike/command/command.rb

```diff
@@ -318,6 +320,73 @@ def set_batch_get(policy, batch_namespace, bin_names, read_attr)
       end_cmd
     end
 
+    def set_batch_index_get(policy,batch, bin_names, read_attr)
+      bin_name_size = 0
+      operation_count = 0
+      field_count = 1
+      if bin_names
+        bin_names.each do |bin_name|
+          bin_name_size += bin_names.bytesize + OPERATION_HEADER_SIZE
+        end
+        operation_count = bin_names.length
+      end
+      begin_cmd
+      @data_offset += FIELD_HEADER_SIZE + 5
```

Comment:
@jhecking  actually i don't know what this number stands for i just replicated the java code so can you please guide what this actually are or can give me link for documentation so that i can name them properly

Reply:
Fair enough. The [wire protocol documentation](https://www.aerospike.com/docs/dev_reference/wire_protocol.html) can be of some help. But it's unfortunately not complete, as it does not describe the layout of the message fields, e.g. the BATCH_INDEX field in particular.

There are mainly 3 constants, that I think need further explanation, on lines 334, 338 and 343. All of them are used to calculate the total size of the BATCH_INDEX field in the message. Naming these constants well is actually really difficult as most of them are the sum of the sizes of one or more elements of that field. Let's settle for just adding some clarifying comments:

Line 334:

    @data_offset += FIELD_HEADER_SIZE + 4 + 1 # batch.keys.length + flags

Line 338:

    @data_offset += key.digest.length + 4 # 4 byte batch offset

Line 343:

    @data_offset += key.namespace.bytesize + FIELD_HEADER_SIZE + 1 + 1 + 2 + 2 # repeat/no-repeat flag + read_attr flags + field_count + operation_count

That's not great but the best I can come up with at the moment.